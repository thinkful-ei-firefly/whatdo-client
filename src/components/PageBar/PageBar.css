.pagination-bar {
  text-align: center;
  margin-top: 20px;
}

.pagination-prev-next-button {
  /* button styles */
  padding: 10px 20px;
  /* color: white; */
  /* border: black 3px solid; */
  border-radius:20px ;
  background-color: #03a9f4;
}

.pagination-prev-next-button:hover{
  cursor: pointer;
}

.pagination-prev-next-button a{
  font-size: 17px;
  color: white;
  height: 100%;
  width: 100%;
}

.pagination-page-num {
  /* page number styles */
  /* font-weight: bold; */
  background-image: radial-gradient(#487eb0da, rgb(255, 255, 255));
  border-radius:20px ;
  padding: 10px 20px;
  margin: 10px;
}

.bold{
  font-weight: bold;
}
