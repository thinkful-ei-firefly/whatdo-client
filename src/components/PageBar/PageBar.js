import React from 'react'
import SearchContext from '../../contexts/SearchContext'
import './PageBar.css'

export default class PageBar extends React.Component {
  state = { loading: false }

  static contextType = SearchContext

  asyncSetState = (state) => {
    return new Promise((resolve) => {
      this.setState(state, resolve)
    });
  }

  handlePagePrev = async () => {
    this.asyncSetState({loading: true})
      .then(res => this.context.prevPage())
      .then(res => this.setState({loading: false}))
  }

  handlePageNext = async () => {
    this.asyncSetState({loading: true})
      .then(res => this.context.nextPage())
      .then(res => this.setState({loading: false}))    
  }

  render () {
    return (
      <div className="pagination-bar">
        <button 
          className="pagination-prev-next-button" 
          disabled={this.state.loading}
          onClick={this.handlePagePrev}
        >
          <a href="#scroll-target">{'<< Prev'}</a>          
        </button>

        <span className="pagination-page-num">  -  Page {this.context.pageNum} of {this.context.pageCount}  -  </span>

        <button 
          className="pagination-prev-next-button" 
          disabled={this.state.loading}
          onClick={this.handlePageNext}
        >
          <a href="#scroll-target">{'Next >>'}</a>
        </button>
      </div>
    );
  }
}