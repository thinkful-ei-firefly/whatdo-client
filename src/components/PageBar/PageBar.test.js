import React from 'react'
import PageBar from './PageBar'
import renderer from 'react-test-renderer'

describe('<PageBar />', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(
        <PageBar />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})