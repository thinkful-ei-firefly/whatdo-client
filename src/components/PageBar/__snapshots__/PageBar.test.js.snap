// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageBar /> renders as expected 1`] = `
<div
  className="pagination-bar"
>
  <button
    className="pagination-prev-next-button"
    disabled={false}
    onClick={[Function]}
  >
    <a
      href="#scroll-target"
    >
       Prev 
    </a>
  </button>
  <span
    className="pagination-page-num"
  >
        Page 
    <span
      className="bold"
    >
      1
    </span>
     of 
    1
       
  </span>
  <button
    className="pagination-prev-next-button"
    disabled={false}
    onClick={[Function]}
  >
    <a
      href="#scroll-target"
    >
      Next
    </a>
  </button>
</div>
`;
